### ISSUE TITLE
Multicol* add extra leftmargin in the second column

### ISSUE THREAD

**Issue by l-atome** [latex3/latex2e#806]
## Brief outline of the bug

the result shows a left margin in the right column

Excuse me, the code is not so beautiful but has the advantage to reproduce the bug

Thanks for all your work 

## Minimal example showing the bug

```latex
\RequirePackage{latexbug}
\documentclass{book}
\usepackage{multicol}

\parskip = 0.2 cm
\parindent = 0 cm
%% Donnée pour multicol
  \setlength{\columnsep}{30pt}
  \setlength{\columnseprule}{1pt}





\begin{document}

sdfg

sdfg

\begin{multicols*}2


exercice
On considère le pseudo-code suivant :
L'exécution de ce code affichera :




exercice

On considère le pseudo-code suivant :


Donner la valeur affichée lors de l'exécution de cet algorithme. \hfill



exercice

 oooooooooo
S\tracingmacros1
%\columnbreak sd

Dans le plan et avec le repère représenté ci-dessous, on considère les $7$ points ci-dessous :

qsdf

qsdf

qsdf

qsfd

qsdf

qsf

qsdf

qsdf

qsdf

qsfd

qsdf

qsfd

qsdf

qsf

qsdf

qsdf

qsdf

qsfd

qsdf

qsdf

qsdf

qsfd

qsdf

qsfd

qsdf

qsf

qsdf

qsdf

qsdf

qsfd

On considère les trois fonctions suivantes prenant pour argument un point :


Les points qui, appliqués à la fonction {\tt verif()}, renverront {\tt Vrai} sont :

exercice

Dans le plan et avec le repère représenté ci-dessous, on considère les $7$ points ci-dessous :




\vbox to9cm{qsdf}


{{\hsize2cm\hbox to1cm{aaaaa\hss}}}

sdfg

sdgf

sdg

sdgf

sdfg

qsf

qsdf


Les points qui, appliqués à la fonction {\tt verif()}, renverront {\tt Vrai} sont :

\end{multicols*}
\end{document}
```

## Log file (required) and possibly PDF file  

[aTravailler-cs-5-13.log](https://github.com/latex3/latex2e/files/8537080/aTravailler-cs-5-13.log)
[aTravailler-cs-5-13.pdf](https://github.com/latex3/latex2e/files/8537081/aTravailler-cs-5-13.pdf)

**Comment by FrankMittelbach** [comment]
There is no bug. You are using `\hsize` in the middle of your main text and you change its value. Altering hsize in the middle of documents is not supported LaTeX which assumes that `\hsize` has a fixed value for the current page.
Perhaps you think that
```
{{\hsize2cm\hbox to1cm{aaaaa\hss}}}
```
means that the change to hsize is local to that box but this is just a misunderstanding how TeX works.
After
```
{{\hsize2cm
```
the value of hsize is 2cm and the following `\hbox` makes TeX realize that it is time to enter the output routine. So the OR is entered with a value of `\hsize=2cm`. But the OR assumes that hsize represents the width of the output box it receives. As this no longer is the case your coding confuses the whole output routine processing.

In other words, if you want to make boxes with special horizontal sizes you need to use either proper LaTeX commands like `\parbox` or you need to change `\hsize` inside the box but not on the outside as part of the main galley because groups do not help you there if the output routine is triggered in the middle of your `{{...}}`

**Comment by l-atome** [comment]
Really thanks for the answer and the very clear explanation
Good day
Tom
